packages/react-inbox/src/components/Messages2.0/Message.tsx

```diff
@@ -194,31 +209,53 @@ const MessageWrapper: React.FunctionComponent<
     return actions[0].href;
   }, [data?.clickAction, actions]);
 
+  const handleClickMessage = (event?: React.MouseEvent) => {
+    event?.preventDefault();
+
+    if (!messageId) {
+      return;
+    }
+
+    if (!read) {
+      markMessageRead(messageId);
+    }
+
+    if (clickAction && courier.onRouteChange) {
+      courier.onRouteChange(clickAction);
+    }
+  };
+
   let containerProps: {
     href?: string;
     onMouseDown?: (event: React.MouseEvent) => void;
     rel?: string;
     target?: string;
-  } = {};
+    tabIndex: number;
+  } = {
+    tabIndex: 0,
+  };
 
   if (clickAction) {
-    containerProps.href = clickAction;
-    containerProps.onMouseDown = handleClickMessage;
+    if (!courier.onRouteChange) {
```

Comment:
@rileylnapier Might be a little cleaner to do: if (clickAction && !courier.onRouteChange) so as to not nest if statements too many levels

Reply:
i would need 2 if statements in that case 

if (clickAction && courier.onRouteChange)

as well as

if (clickAction && !courier.onRouteChange)

so nesting it makes it cleaner in my opinion.  i do agree with you that keeping code all the way to the left is my preference but here id rather keep all the logic related to "clickAction" in one place vs 2 conditionals